npm/ng-packs/packages/core/src/lib/services/internet-connection-service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, OnDestroy, OnInit, VERSION } from '@angular/core';
+import { InternalStore } from '../utils/internal-store-utils';
+import { fromEvent, merge, of, Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface InternetConnectionState{
+  status: boolean;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class InternetConnectionService{
+  private store = new InternalStore<InternetConnectionState>({ status: true });

```

Comment:
initial state might be true or false. There might be a state that name is unkown. We didn't know then initialized